.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,69 @@
+name: build and test
+
+on:
+  push:
```

Comment:
for this PR this is causing an extra run because the branch is on this repo, I have redone my fork and PRs will be from there to avoid confusion, and those should only execute once